### PR TITLE
[TTAHUB-3345] Add filters to QA dashboard, frontend

### PR DESCRIPTION
**Pull request by thewatermethod**
## Description of change
Adds filters to the QA dashboard. The size of this change is mainly due to refactoring - other instances of our ```useFilters``` hook had to be updated. The upside of this is that more logic is now internal to the hook and we have less repeated code on the various dashboards.

This change also incorporated some updates to allow a series of pages to share filters while only surfacing some to the UI depending on the page's configuration.

One page is yet to be coded but once it is, it should be a simple drop-in/configuration to get it added.

Finally, I added an "SSDI" data fetcher that will strip unneeded key/values from the filter object before sending a request to the API. Given the uncertain state of the backend, we may need to refactor that but it will likely still be useful given the shared filters

## How to test
- No data is being fetched from the backend. This is just a test to see if the filters function as expected pre-data-fetching.
- Visit the QA dashboard.
- Add some filters. Visit one of the subpages, change some filters.
- Note that adding a filter persists between pages (if both pages have it) as does deleting.
- Note also that if a page does not have a filter present in another, it disappears on the page that doesn't have it but reappears when you return.

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3345

### Every PR

- [ ] Meets issue criteria
- [ ] JIRA ticket status updated
- [ ] Code is meaningfully tested
- [ ] Meets accessibility standards (WCAG 2.1 Levels A, AA)
- [ ] API Documentation updated
- [ ] Boundary diagram updated
- [ ] Logical Data Model updated
- [ ] [Architectural Decision Records](https://adr.github.io/) written for major infrastructure decisions
- [ ] UI review complete

### Before merge to main

- [ ] OHS demo complete
- [ ] Ready to create production PR

### Production Deploy

- [ ] Staging smoke test completed

### After merge/deploy

- [ ] Update JIRA ticket status

### CODE REVIEW

File: frontend/src/fetchers/ssdi.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
+import join from 'url-join';
+import { get } from './index';
+import { filtersToQueryString } from '../utils';
+import { QA_DASHBOARD_FILTER_CONFIG } from '../pages/QADashboard/constants';
+
+const ssdiUrl = join('/', 'api', 'ssdi');
+
+const allowedTopicsForQuery = {
+  'recipients-with-no-tta': [
```
Comment (AdamAdHocTeam) [inline frontend/src/fetchers/ssdi.js:9]: Not sure if it makes sense to put this in a constants if we will use it in other locations.

File: frontend/src/hooks/useSubFilters.js [hunk 2]
```diff
@@ -0,0 +1,42 @@
+import {
```
Comment (AdamAdHocTeam) [inline frontend/src/hooks/useSubFilters.js:1]: This really serves as a "filter" for allowed filters. Should we call is something like 'useAllowedFilters'.

File: frontend/src/pages/Landing/index.js [hunk 3]
```diff
@@ -303,7 +303,6 @@ function Landing() {
               filters={filters}
               onApplyFilters={onApply}
               onRemoveFilter={onRemoveFilter}
-              // filterConfig={getFilters()}
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/Landing/index.js:306]: Can we remove this? Or will it come into play later?

File: frontend/src/pages/QADashboard/RecipientsWithNoTta/index.js [hunk 4]
```diff
@@ -1,24 +1,54 @@
-/* eslint-disable no-alert */
-/* eslint-disable no-console */
 import React, {
-  useState,
+  useContext,
+  // useState,
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/QADashboard/RecipientsWithNoTta/index.js:3]: Some commented out lines not sure if intentional.

File: frontend/src/pages/QADashboard/RecipientsWithOhsStandardFeiGoal/index.js [hunk 5]
```diff
@@ -1,25 +1,56 @@
...
 import Drawer from '../../../components/Drawer';
 import ContentFromFeedByTag from '../../../components/ContentFromFeedByTag';
 import DrawerTriggerButton from '../../../components/DrawerTriggerButton';
+import FilterPanel from '../../../components/filter/FilterPanel';
+import FilterPanelContainer from '../../../components/filter/FilterPanelContainer';
+import useFilters from '../../../hooks/useFilters';
+import { QA_DASHBOARD_FILTER_KEY, QA_DASHBOARD_FILTER_CONFIG } from '../constants';
+import UserContext from '../../../UserContext';
 
+const ALLOWED_SUBFILTERS = [
```
Comment (AdamAdHocTeam) [inline frontend/src/pages/QADashboard/RecipientsWithOhsStandardFeiGoal/index.js:24]: If we will have a bunch of these scattered should we have in constants file?

Comment (thewatermethod) [reply frontend/src/pages/QADashboard/RecipientsWithOhsStandardFeiGoal/index.js:24]: they are on a per page basis

File: frontend/src/fetchers/ssdi.js (hunk 1 above)
Comment (thewatermethod) [reply frontend/src/fetchers/ssdi.js:9]: I don't think this is final until the backend is done, but sure at that point we can move it into a constant

File: frontend/src/hooks/useSubFilters.js (hunk 2 above)
Comment (thewatermethod) [reply frontend/src/hooks/useSubFilters.js:1]: sure if you want

File: frontend/src/pages/Landing/index.js (hunk 3 above)
Comment (thewatermethod) [reply frontend/src/pages/Landing/index.js:306]: this line is me removing it

File: frontend/src/pages/QADashboard/RecipientsWithNoTta/index.js (hunk 4 above)
Comment (thewatermethod) [reply frontend/src/pages/QADashboard/RecipientsWithNoTta/index.js:3]: I'll delete

**Review by AdamAdHocTeam — COMMENTED** [review]
Probably unrelated but noticed a z index issue on the resource dashbaord when the selected filter list grows over the first widget.

image

**Review by AdamAdHocTeam — APPROVED** [review]
Overall looks really good.

Left minor comments (and z-index issue).

**Pull request merged**